src/explore-education-statistics-common/src/validation/serverValidations.ts

```diff
@@ -132,44 +132,76 @@ export function convertServerFieldErrors<FormValues>(
   );
 }
 
-export function isServerValidationError(
+/**
+ * Asserts whether the given error object is an
+ * {@link AxiosError<ServerValidationErrorResponse<T>>}.
+ *
+ * @param error - the error object to check the type of.
+ */
+export function isServerValidationError<T extends string = string>(
   error: unknown,
-  errorMessage?: string,
-): error is AxiosError<ServerValidationErrorResponse> {
+): error is AxiosError<ServerValidationErrorResponse<T>> {
   if (!isAxiosError(error) || !error.response?.data) {
     return false;
   }
 
   const errorDataAsValidationError = error.response
-    .data as ServerValidationErrorResponse;
+    .data as ServerValidationErrorResponse<T>;
 
-  const isServerError =
+  return (
     errorDataAsValidationError.errors !== undefined &&
     errorDataAsValidationError.status !== undefined &&
-    errorDataAsValidationError.title !== undefined;
+    errorDataAsValidationError.title !== undefined
+  );
+}
 
-  if (!errorMessage) {
-    return isServerError;
+/**
+ * Asserts whether the given error object is an
+ * {@link AxiosError<ServerValidationErrorResponse<T>>}.
+ *
+ * See {@link isServerValidationError}
+ *
+ * Additionally, this method checks whether or not the error contains any of
+ * the error messages provided, using {@param fieldName} to determine field
+ * validation, or global validation if {@param fieldName} is omitted.
+ *
+ * If any are included in this error object, this method will return true.
```

Comment:
Rather than lead with telling about the type assertion, would suggest making the focus of this docblock about the method checking that the error response contains a specific message (see below's comment about changing the name to `hasErrorMessages` as well.

Reply:
Yup, removed the type checking altogether as suggested, in favour of checking type ahead of time with existing function.